src/pages/index.js

```diff
@@ -51,37 +63,140 @@ const profileEditModal = new PopupWithForm(
   ".profile__title",
-  ".profile__description"
-);
-
-const cardSection = new Section(
-  { items: initialCards, renderer: createCard },
-  ".cards__list"
+  ".profile__description",
+  ".profile__image"
 );
 
 //functions
 
 function handleProfileFormSubmit(formData) {
-  console.log(formData);
-  profileUserInfo.setUserInfo(formData.title, formData.description);
+  //fetch request to change the name and description on the server
+  profileEditModal.setButtonText("Saving...");
+  //profileEditModal.reset();
+  api
+
+    .updateUserInfo(formData)
+    .then((res) => {
+      profileUserInfo.setUserInfo(res.name, res.about);
+      profileEditModal.close();
+      //modal__form.reset();
+    })
+
+    .catch((err) => {
+      console.error(err);
+      alert("Failed to change user info");
+      console.log(formData);
+    })
+    .finally(() => {
+      profileEditModal.setButtonText("Save");
+    });
 }
 
 function handleCardAddSubmit(inputValues) {
   const cardData = { name: inputValues.title, link: inputValues.url };
-  cardSection.addItem(cardData);
   addFormValidator.resetValidation();
+  addCardModal.setButtonText("Saving...");
+  api
+    .addCard(cardData)
+    .then((res) => {
+      const cardElement = createCard(res);
+      cardSection.addItem(cardElement);
+      addCardModal.close();
+    })
+    .finally(() => {
+      addCardModal.setButtonText("Save");
+    });
+}
+
+function handleLikeClick(card) {
+  console.log("like card is running");
+  // call the api and pass card.id and card.isLiked
+  api
+    .updateCardLike(card.id, card.isLiked)
+    .then(() => {
+      card.addLike();
+      //card.isLiked = true;
+
+      //card.setLikeButtonState();
+    })
+    .catch((err) => {
+      console.error(err);
+      alert(`${err} Failed to like card`);
+    });
+
+  // after api is successful, we can then update the UI
+}
+
+function handleUnlikeClick(card) {
+  console.log("unlike card is running");
+  api
+    .updateCardLike(card.id, card.isLiked)
+    .then(() => {
+      card.removeLike();
+      //card.isLiked = false;
+
+      //card.setLikeButtonState();
+    })
+    .catch((err) => {
+      console.error(err);
+      alert(`${err} Failed to unlike card`);
+    });
 }
 
 function handleImagePreview(name, link) {
   previewImageModal.open(name, link);
 }
+//runs when clicking the trash button
+function handleDeleteClick(card) {
+  //runs when clicking the yes button
+  function deleteCard() {
+    //delete the card on the server.
+    deleteConfirmationModal.setButtonText("Delete");
+    api
+      .deleteCard(card.id)
+      .then(() => {
+        //closes modal
+        deleteConfirmationModal.close();
+        //deletes card on the UI
+        card.handleDeleteCardLocally();
+      })
+      .catch((err) => {
+        console.error(err);
+        alert(`${err} Failed to delete card`);
+      })
+
+      .finally(() => {
+        deleteConfirmationModal.setButtonText("Deleting...");
```

Comment:
Needs correction: Here you should end the loading (make the button text initial)